frontend/src/features/directory/components/FilterableVendorTable.tsx

```diff
@@ -76,6 +77,47 @@ function FilterableVendorTableContent({ uniqueRegions, tags, vendors, favoriteVe
     return result;
   }, [searchQuery, filteredVendors, sortOption]);
 
+  const [prevParams, setPrevParams] = useState<string | null>(null);
+
+  useEffect(() => {
+    const currentParams = searchParams.toString();
+  
+    if (prevParams !== null && currentParams !== prevParams) {
+      const hasSearchChanged = searchQuery !== searchParams.get(SEARCH_PARAM) || "";
```

Comment:
do we not need a parenthesis?

searchQuery !== (searchParams.get(SEARCH_PARAM) || "")

Reply:
good catch!